lib/extensions/attachable/active_record/base.rb

```diff
@@ -27,6 +41,51 @@ def files=(files)
         attachment_references.build(file: file)
       end
     end
+
+    private
+
+    # Delete the attachment references which are not in the content.
+    def update_attachment_references
+      return if attachment_references.empty?
+
+      attachment_reference_ids = active_attachment_reference_ids
```

Comment:
They attachment without an id won't be marked_for_destruction.  I meant if we do this by assigning   an array of references, we need first to construct the array, and remove the deleted attachments from the array, not very cleaner to me.